app/script/search/SearchRepository.js

```diff
@@ -56,6 +61,52 @@ z.search.SearchRepository = class SearchRepository {
     this.logger = new z.util.Logger('z.search.SearchRepository', z.config.LOGGER.OPTIONS);
   }
 
+  /**
+   * Search for a user in the given user list and given a search term.
+   * Doesn't sort the results and keep the initial order of the given user list.
+   *
+   * @param {string} term - the search term
+   * @param {Array<z.entity.User>} userEntities - entities to match the search term against
+   * @param {Array<z.search.SearchRepository.CONFIG.SEARCHABLE_FIELDS>} properties=['first_name', 'last_name', 'username'] - list of properties that will be matched against the search term
+   *    the order of the properties in the array indicates the priorities by which results will be sorted
+   * @returns {Array<z.entity.User>} the filtered list of users
+   */
+  searchUserInSet(term, userEntities, properties) {
+    const SEARCHABLE_FIELDS = z.search.SearchRepository.CONFIG.SEARCHABLE_FIELDS;
+    properties = properties || [SEARCHABLE_FIELDS.FIRST_NAME, SEARCHABLE_FIELDS.LAST_NAME, SEARCHABLE_FIELDS.USERNAME];
+    const excludedEmojis = Array.from(term).filter(char => {
+      return z.util.EmojiUtil.UNICODE_RANGES.includes(char);
+    });
+
+    const matches = (userEntity, property, fromStart) => {
+      const value = ko.unwrap(userEntity[property]);
+      return z.util.StringUtil.compareTransliteration(value, term, excludedEmojis, fromStart);
+    };
+
+    const weightedResults = userEntities.reduce((results, userEntity) => {
+      const matchedProperties = properties.filter(property => matches(userEntity, property, false));
+
+      if (!matchedProperties.length) {
+        return results;
+      }
+
+      // add weight to the result based on the properties that matched and the position of the property in the property list
+      const weight = matchedProperties.reduce((weightValue, property) => {
+        const propertyImportance = properties.reverse().indexOf(property);
+        return weightValue + propertyImportance;
+      }, 0);
```

Comment:
I think the reversing should not be done in the set. This is part of the visualization logic that we won't need for other applications of this search capability.

Reply:
Not sure I get your point. The point of the reverse here is to get the importance of the field that matched in order to assign a weight to a search result. This has nothing to do with visualization (well we do sort in the end, that is true)